src/composables/useTransactions.ts

```diff
@@ -311,6 +312,11 @@ export default function useTransactions() {
       status: 'pending',
     };
 
+    if (Object.keys(transactionsMap).length > MAX_CACHED_TRANSACTIONS) {
+      const keys = Object.keys(transactionsMap);
```

Comment:
This should be sorted by addedTime, otherwise it will delete the first one alphabetically I believe.